### DISCUSSION TITLE
Ability to get the latest file from an external call

### DISCUSSION THREAD

**Discussion by 0xperp** [observablehq/framework#1426] [Q&A] [answered]
Hi All - 

I know that this maybe isn't the best use case (I should probably have more of an API), but I load `.json` files into my site that are then pulled in via `/_file/data/<file-name>.<content-hash>.json`. These are rebuilt every hour on a cron job so the content hash changes quite frequently

I would like to be able to query the file from an external site but it seems like there is no way of knowing the content hash, listing out all the data files and finding the most recent, or something like that? Maybe I am missing something, but would be very much appreciated if this is something that can be done! Thanks.

**Comment by Fil** [comment, marked as answer]
Currently there is no "site map" or "files manifest", and the only way to do this is to find a page that references them, and scrape it to isolate the following block:

```js
registerFile("./hello.txt", {"name":"./hello.txt","mimeType":"text/plain","path":"./_file/hello.89e049c5.txt","lastModified":1717496456910});
registerFile("./horse.jpg", {"name":"./horse.jpg","mimeType":"image/jpeg","path":"./_file/horse.2d33a223.jpg","lastModified":1717496456910});
registerFile("./pryvit.txt", {"name":"./pryvit.txt","mimeType":"text/plain","path":"./_file/pryvit.a1bbae9b.txt","lastModified":1717496457010});
registerFile("./volcano.json", {"name":"./volcano.json","mimeType":"application/json","path":"./_file/volcano.f5223ec9.json","lastModified":1717496457010});
```

**Reply by 0xperp** [reply]
Great thanks! Got it with registering all files at `/data/raw` and a script like

```
fetch('localhost:3000/data/raw')
  .then(response => response.text())
  .then(data => {
    const fileLines = data.split('\n');
    const filteredFiles = fileLines
      .filter(line => line.includes('"path"'))
      .map(line => {
        const pathStartIndex = line.indexOf('"path":"') + 8;
        const pathEndIndex = line.indexOf('"', pathStartIndex);
        const filePath = line.substring(pathStartIndex, pathEndIndex);
        return `${filePath}`;
      });

    const filteredData = { files: filteredFiles };
    console.log(filteredData);
  })
  .catch(error => console.error('Error fetching data:', error));
```

Appreciate the help
